routing/index_graph.cpp

```diff
@@ -0,0 +1,197 @@
+#include "index_graph.hpp"
+
+#include "base/assert.hpp"
+#include "base/exception.hpp"
+
+namespace routing
+{
+IndexGraph::IndexGraph(unique_ptr<GeometryLoader> loader, shared_ptr<EdgeEstimator> estimator)
+  : m_geometry(move(loader)), m_estimator(move(estimator))
+{
+  ASSERT(m_estimator, ());
+}
+
+void IndexGraph::GetOutgoingEdgesList(JointId jointId, vector<TEdgeType> & edges) const
+{
+  GetEdgesList(jointId, edges, true);
+}
+
+void IndexGraph::GetIngoingEdgesList(JointId jointId, vector<TEdgeType> & edges) const
+{
+  GetEdgesList(jointId, edges, false);
+}
+
+double IndexGraph::HeuristicCostEstimate(JointId jointFrom, JointId jointTo) const
+{
+  return m_estimator->CalcHeuristic(GetPoint(jointFrom), GetPoint(jointTo));
+}
+
+m2::PointD const & IndexGraph::GetPoint(JointId jointId) const
+{
+  return m_geometry.GetPoint(GetFSeg(jointId));
+}
+
+void IndexGraph::Export(vector<Joint> const & joints)
+{
+  m_fsegIndex.Import(joints);
```

Comment:
For the time being Import is called from Export. Probably it's worth renaming IndexGraph::Export() to IndexGraph::Import() as well.